partcad/src/partcad/cache_hash.py

```
@@ -0,0 +1,121 @@
+        self.name = name
+        self.is_empty = True
+        self.is_used = False
+        if not user_config.cache:
+            # Caching is disabled, no initialization needed
+            self.hasher = None
+            return
+
+        if hasher != None:
+            self.hasher = hasher.copy()
+        elif algo == "md5":
+            self.hasher = hashlib.md5()
```

Ruff (0.8.2)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)